source/memory-store.ts

```diff
@@ -30,16 +24,33 @@ export default class MemoryStore implements Store {
 	windowMs!: number
 
 	/**
-	 * The map that stores the number of hits for each client in memory.
+	 * These two maps store usage (requests) and reset time by key (IP)
+	 *
+	 * They are split into two to avoid having to iterate through the entire set to determine which ones need reset.
+	 * Instead, Clients are moved from previous to current as new requests come in.
+	 * Once windowMS has elapsed, all clients left in previous are known to be expired.
+	 * At that point, the cache pool is filled from previous, and any remaining clients are cleared.
+	 *
 	 */
-	hits!: {
-		[key: string]: number | undefined
-	}
+	previous = new Map<string, Client>()
+	current = new Map<string, Client>()
 
 	/**
-	 * The time at which all hit counts will be reset.
+	 * Cache of unused clients, kept to reduce the number of objects created and destroyed.
+	 *
+	 * Improves performance, at the expense of a small amount of RAM.
+	 *
+	 * Each individual entry takes up 152 bytes.
+	 * In one benchmark, the total time taken to handle 100M requests was reduced from 70.184s to 47.862s (~32% improvement) with ~5.022 MB extra RAM used.
```

Comment:
Should we add this benchmark to the repo as `test/benchmark/memory-store.benchmark.ts`?

Reply:
No, at least not in its current state. It's pretty ad hoc, and coupled to the implementation. I was making changes to the testing script as I changed the implementation details, so that's not even the current iteration.

Also it's very dependent on the machine you run it on, so I don't think it would be very useful to compare across machines.

Reply:
Oh alright then